apps/remix-ide/src/app/compiler/compiler-utils.js

```diff
@@ -25,8 +25,10 @@ export function canUseWorker (selectedVersion) {
   const version = semver.coerce(selectedVersion)
   const isNightly = selectedVersion.includes('nightly')
   return browserSupportWorker() && (
+    // All compiler versions (including nightlies) after 0.6.3 are wasm compiled
     semver.gt(version, '0.6.3') ||
-    (semver.gt(version, '0.3.6') && !isNightly)
+    // Only releases are wasm compiled starting with 0.3.6
```

Comment:
@cameel can you confirm these two comments? :)

Reply:
We have wasm builds for all releases starting with 0.3.6, [except for 0.5.17](https://github.com/ethereum/solidity/issues/10329).

Reply:
then we should include 0.5.17 too in the check.
Please check manually for 0.3.6 as I remember when I was adding this it has a problem of `Uncaught JavaScript exception: RangeError: Maximum call stack size exceeded`

Reply:
Well, I did test both wasm and non-wasm 0.3.6 back when I was doing the comparison of generated bytecode (between binaries of a given version on all platforms) in the CI. That was using node.js. I did not get errors like this one so this must be a browser thing.

About the check: not sure if it's easy to do that here, but the simplest one would be to assume that whatever is in the `wasm/` dir needs the worker. We do plan to add a wasm build of 0.5.17 eventually so it's better if it's not excluded permanently.

Reply:
```
contract C {
    struct S {
        S[] arr;
    }

    S s;

    function test() {
        s.arr.length = 10;
    }
}
```

Reply:
if you check with the code above, we still have Uncaught JavaScript exception: RangeError: Maximum call stack size exceeded. this was fixed when we moved to worker load of wasm builds. this one was an exception and was still throwing the exception so I left it out. I'm not sure why we have errors on loading now

Reply:
but I guess this is not important as it is not working at all.

Reply:
Ok this does fail but it seems to be a bug in the compiler or solc-js rather than it not being wasm. I just tried to compile that snippet locally using `solc-js` (`./solcjs test.sol --bin`) and it crashes this way on all versions between 0.3.6 and 0.4.8, both asm.js and wasm ones.